### PR TITLE
Bug 482684 - ConfigDescriptionParameter methods should not return null

### PR DESCRIPTION
**Pull request by chriskn**
Changes:
ConfigDescriptionParameter.java contains only boolean attributes (no
Boolean Attributes anymore). Null safe conversion from Boolean to boolean in constructor

Implemented ConfigDescriptionParameterBuilderTest.groovy to clarify usage of ConfigDescriptionParameterBuilder.java and ConfigDescriptionParameter.java

Added default constructor to ConfigDescriptionParameter.java

Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=482684

Signed-off-by: Christoph Knauf  <christoph.knauf@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=482684

**Comment by maggu2810** [comment]
LGTM

@kaikreuzer: The guidelines state that the 
```text
Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=482684
```
line have to be part of the commit message.

Is there something changed?
There has been a PR a few days ago, you and Dennis posted something about that (Bug number in front of commit message, etc.).

But I think we should be conform to the Eclipse Guidelines.
Correct?

File: bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigDescriptionParameter.java [hunk 1]
```diff
@@ -183,28 +192,48 @@ public ConfigDescriptionParameter(String name, Type type, BigDecimal minimum, Bi
...
         this.context = context;
-        this.required = required;
         this.defaultValue = defaultValue;
         this.label = label;
         this.description = description;
+        this.multipleLimit = multipleLimit;
 
-        if (options != null)
+        if (readOnly == null) {
+            this.readOnly = false;
```
Comment (kaikreuzer) [inline bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigDescriptionParameter.java]: This is imho not fully clean. In the noargs constructor you keep the `this.readonly` uninitialized while you specifically set it here to false. I would prefer to have both constructors operate the same way. The easiest is probably to explicitly initialize the field in the class like `boolean readOnly = false;` and only keep the `!=` case here in the constructor. Same for the other booleans.

**Comment by chriskn** [comment]
Changes according to comments

**Comment by kaikreuzer** [comment]
Thanks!

**Comment by maggu2810** [comment]
If I understand the Eclipse Guidelines correctly the "Bug" entry in the commit message should be part of the footer.

<pre>... a link to the Bug ... should be included in the message footer...</pre>

as shown in the example of https://wiki.eclipse.org/Development_Resources/Contributing_via_Git
```text
commit 862e6ff22ad56c10df6de3385ffa4c7d02363d1d
Author: Joe Somebody <somebody@someplace.net>
Date:   Mon Jun 17 17:19:38 2013 -0700

    [410937] Auto share multiple projects in single job
    
    When multiple projects are imported together, perform all the necessary
    auto shares in a single job rather than spawning a separate job for each
    project.
    
    Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=410937
    Also-by: Some Otherperson <otherperson@someplace.net>
    Signed-off-by: Joe Somebody <somebody@someplace.net>
```

**Comment by maggu2810** [comment]
@chriskn Thanks :wink:

**Pull request merged**